calico-vpp-agent/cni/cni_server.go

```diff
@@ -378,9 +405,32 @@ func (s *Server) ServeCNI(t *tomb.Tomb) error {
 	}
 
 	pb.RegisterCniDataplaneServer(s.grpcServer, s)
+
+	netsSynced := make(chan bool)
+	nets := make(map[string]*watchers.NetworkDefinition)
+	go func() {
+		for t.Alive() {
+			event := <-s.cniMultinetEventChan
+			switch event.Type {
+			case common.NetsSynced:
+				nets = event.New.(map[string]*watchers.NetworkDefinition)
+				s.networkDefinitions = nets
+				netsSynced <- true
+			case common.NetAdded:
+				netDef := event.New.(*watchers.NetworkDefinition)
+				s.networkDefinitions[netDef.Name] = netDef
```

Comment:
There is potential race-condition on `networkDefinitions` as it is accessed by two goroutines (this one and the CNI calls)
A way around this would be to `s.lock.Lock()` & `s.lock.Unlock()` around the accesses to it as it is also taken by `AddVppInterface()` and `DelVppInterface()`